### PR TITLE
Migrate Layout Basics Codelab to Null Safety

### PR DESCRIPTION
**Pull request by miquelbeltran**
@Sfshaza @RedBrogdon 

In this PR I have migrated the Layout Basics codelab following the steps in the migration plan document.

The PR is split in commits in a way that hopefully makes easier to follow the changes!

Improvements:
- Dartpads are now embedded in markdown rather than on gists.
- Code is migrated to null safety, and it is enabled by default on the dartpads. All changes happen in the test code almost.
- Upgraded text styles to the current material design spec in the "Putting it all together" chapter.

Live version here: https://mb-dart-dev-1.firebaseapp.com/docs/codelabs/layout-basics_migrated

The "Name non-constant identifiers using lowerCamelCase" warning that appears is an issue on DartPad. I have already reported to Andrew.

Please feel free to edit anything as necessary before merging :)

### CODE REVIEW

**Review by sfshaza2 — COMMENTED** [review]
It LGTM, @miquelbeltran! But I'll let @RedBrogdon also take a look.

File: src/docs/codelabs/layout-basics_migrated.md
Comment (RedBrogdon) [inline src/docs/codelabs/layout-basics_migrated.md:2166]: In general, we always try to use trailing commas to keep things indented by two characters, in part because DartPad's width is so constrained when shown in an iframe.

Here, if there was a comma in between the two parentheses at the end of the line, the last one would be moved to the next line, and both parameters for `Padding` would be indented two character instead of four.

File: src/docs/codelabs/layout-basics_migrated.md
Comment (RedBrogdon) [inline src/docs/codelabs/layout-basics_migrated.md:2383]: Here as well, a comma after `headline5` will help with indenting.

File: src/docs/codelabs/layout-basics_migrated.md
Comment (RedBrogdon) [inline src/docs/codelabs/layout-basics_migrated.md:55]: `run-true` should be taken out here and in the others. Right now it's causing every DartPad embed on the page to execute at the same time, which is a CPU hit for both the browser and our compilation backend. 😄

**Review by RedBrogdon — COMMENTED** [review]
The updated code looks great. My only nits are about indenting and the injection code itself. I'm happy to see these stop relying on gists as well!

File: src/docs/codelabs/layout-basics_migrated.md
Comment (miquelbeltran) [reply src/docs/codelabs/layout-basics_migrated.md]: Ouch! I see, removed now.

File: src/docs/codelabs/layout-basics_migrated.md
Comment (miquelbeltran) [reply src/docs/codelabs/layout-basics_migrated.md:2166]: Agreed, I think this was present in the original Gist, I will fix it now before merge.

**Review by sfshaza2 — COMMENTED** [review]
You rock, @miquelbeltran! @RedBrogdon, I'd merge this now, but it's failing code check. Is the null safe code triggering that?  https://travis-ci.org/github/flutter/website/jobs/756691978

**Comment by miquelbeltran** [comment]
I see, looks like the CI check that extracts and analyzes code snippets in md files does not like the embedded codelab code :) I will research why, but if you know how to fix it please let me know!

**Comment by miquelbeltran** [comment]
I think it may be because the dartpad code contains a section separating the main/solution/test and the code analyzer doesn't like that. As far as I understand, adding `
` over each dartpad snippet would skip the checks. Is that correct?

**Review by RedBrogdon — APPROVED** [review]

**Comment by RedBrogdon** [comment]
LGTM. I've restarted the travis build, which had previously errored out.

**Comment by sfshaza2** [comment]
@RedBrogdon has fixed the cookbook errors in another PR, so I'm forcing a merge.

**Pull request merged**
